tests/bluetooth/ctrl_isoal/src/sub_sets/isoal_test_tx.c

```diff
@@ -4396,8 +4396,501 @@ ZTEST(test_tx_framed, test_tx_framed_2_sdu_3_frag_4_pdu)
  */
 ZTEST(test_tx_framed, test_tx_framed_2_sdu_3_frag_4_pdu_padding)
```

Comment:
There seems to be a lot of very similar or duplicated code blocks in this function. Have you considered implementing a smaller helper function to reduce the number of duplicated lines?

Reply:
I did consider it, but the most volume is in the parameters of the macros for PDU allocation, writing and emit tests. I don't see an improvement if these are further hidden behind functions that try to permute the possibilities. Whether there is allocation, writing or emitting depends very much on what is going on at each step and what the test is trying check at that point.

Reply:
Fair :)